src/drake/core.clj

```diff
@@ -176,6 +181,23 @@
       :vars vars
       :opts (if-not method opts (merge (:opts method) opts)))))
 
+(defn- make-input-stats
```

Comment:
IMO the parser already does too much stuff unrelated to parsing. It shouldn't be reading the filesystem to decide what is meant by the stuff it parses. So where you've put this looks right to me.